presto-hudi/src/test/java/com/facebook/presto/hudi/AbstractHudiDistributedQueryTestBase.java

```diff
@@ -0,0 +1,299 @@
+
+    /**
+     * List of tables present in the test resources directory.
+     * used to test dataskipping/partition prune.
+     */
+    protected static final String HUDI_SKIPPING_TABLE = "data_partition_prune";
+    protected static final String HUDI_SKIPPING_TABLE_NON_HIVE_STYLE = "data_partition_prune_non_hive_style_partition";
+    protected static ConnectorSession connectorSession;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return createHudiQueryRunner(ImmutableMap.of());
+    }
+
+    @AfterClass
+    public void deleteTestHudiTables()
+    {
+        QueryRunner queryRunner = getQueryRunner();
+        if (queryRunner != null) {
+            // Remove the test hudi tables from HMS
+            metastore.dropTable(METASTORE_CONTEXT, HUDI_SCHEMA, HUDI_SKIPPING_TABLE, false);
+            metastore.dropTable(METASTORE_CONTEXT, HUDI_SCHEMA, HUDI_SKIPPING_TABLE_NON_HIVE_STYLE, false);
+        }
+    }
+
+    protected static String getTablePath(String tableName, Path dataDir)
+    {
+        return "file://" + dataDir.resolve(tableName);
+    }
+
+    private static DistributedQueryRunner createHudiQueryRunner(Map<String, String> extraProperties)
+            throws Exception
+    {
+        Session session = testSessionBuilder()
+                .setCatalog(HUDI_CATALOG)
+                .setSchema(HUDI_SCHEMA)
+                .setCatalogSessionProperty(HUDI_CATALOG, "hudi_metadata_table_enabled", "true")
+                .setConnectionProperty(new ConnectorId("hudi"), "hudi_metadata_table_enabled", "true")
+                .build();
+
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(session)
+                .setExtraProperties(extraProperties)
+                .build();
+
+        // setup file metastore
+
+        Path catalogDirectory = queryRunner.getCoordinator().getDataDirectory().resolve("catalog");
+
+        metastore = createFileHiveMetastore(catalogDirectory.toString());
+
+        // create database
+        Database database = Database.builder()
+                .setDatabaseName(HUDI_SCHEMA)
+                .setOwnerName(OWNER_PUBLIC)
+                .setOwnerType(PrincipalType.ROLE)
+                .build();
+        metastore.createDatabase(METASTORE_CONTEXT, database);
+
+        Path testingDataDir = queryRunner.getCoordinator().getDataDirectory().resolve("data");
+
+        try (InputStream stream = Resources.getResource("hudi-skipping-schema-data.zip").openStream()) {
+            unzip(stream, testingDataDir);
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+
+        // Create the test hudi tables for dataSkipping/partition prune in HMS
+        registerHudiTableInHMS(HoodieTableType.COPY_ON_WRITE, HUDI_SKIPPING_TABLE, testingDataDir, Streams.concat(HUDI_META_COLUMNS.stream(), DATA_COLUMNS.stream()).collect(Collectors.toList()));
```

Comment:
I believe the tests only cover CoW table type. Let us add for MoR table type as well?